### ISSUE TITLE
Desktop entry spam on Plasma

### ISSUE THREAD

**Issue by Tatsh** [lsp-plugins/lsp-plugins#376]
#285 describes the same issue for GNOME.

There is a fix that can be applied to Plasma by adding `NoDisplay=true` to every .desktop file. This is [a standard key](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys) unlike the GNOME one. GNOME should respect `NoDisplay=true` too but I cannot confirm.

Do these desktop entry files have to exist for any app to recognise the plugins? From a usability perspective, adding `NoDisplay=true` without having mime types to associate or some other reason makes the .desktop file practically useless, unless the intent is to _not_ have the user launch the plugin separately from the hosting application.

**Comment by sadko4u** [comment]
> NoDisplay means "this application exists, but don't display it in the menus". This can be useful to e.g. associate this application with MIME types, so that it gets launched from a file manager (or other apps), without having a menu entry for it (there are tons of good reasons for this, including e.g. the netscape -remote, or kfmclient openURL kind of stuff). 

This is not right solution because otherwise we won't have `LSP Plugins` menu with it's subitems:

![изображение](https://github.com/lsp-plugins/lsp-plugins/assets/12834464/f975e4bb-8aca-4d30-b554-b2b1f4bcbfee)

I think just `NotShowIn` should be extended to avoid showing in some unexpected places.

**Comment by Tatsh** [comment]
It really depends on what the intent is of these .desktop files.

I use the option _Flatten sub-menus to a single level_:

![image](https://github.com/lsp-plugins/lsp-plugins/assets/724848/4e7e7463-5763-435f-a944-15094563dd37)

With this setting, it's a lot of noise in the Multimedia directory.

And I do not like to customise my menus with KDE's editor because that saves .desktop files in `~/.local/share/applications` that override and then you lose potential updates because of your overrides.

**Comment by sadko4u** [comment]
This is a good reason to understand how 'Flatten' actually works. If it adds all items to some group like 'Plasma' then it is a good reason just to add 'Plasma' to 'NotShownIn' property.

**Comment by Tatsh** [comment]
All flatten does is make the list have a single hierarchy after the category. From:

![image](https://github.com/lsp-plugins/lsp-plugins/assets/724848/9d77fd2d-493b-430d-ab2d-32a627f6f18c)


To:

![image](https://github.com/lsp-plugins/lsp-plugins/assets/724848/af23506b-7e25-4fe9-98f2-b7877c807e48)

**Comment by RoestVrijStaal** [comment]
I'm glad I'm not the only person on this planet who suffers from this kind of "SPAM" as well. I'm running MX 23.3 Libretto KDE

**Why are there .desktop files anyway???** 
What's their use case? At most of the time, the plugins are utilized by "plugin-hosts"  like Audacity, LMMS, PulseEffects / EasyEffects. Why giving those plugins a separate desktop file when the plugins are pure installed as automatic dependency?

Please consider to package those desktop files in a separate package. Or create a new `-headless` package without those evil desktop files. 
Or another way around nuke the desktop files from the original package and create a `-full` or `-standalone` package.

**Comment by sadko4u** [comment]
Just don't install standalone JACK plugins and your problem will be gone.
I don't know how Linux distibution maintainers are handling packages, so I think it is much better if you asked them.
For example, Arch Linux maintainers split LSP plugins into separate packages depending on the plugin format.

**Comment by Tatsh** [comment]
Simple solution on Gentoo: for `media-libs/lsp-plugins`, set `USE=-X`. It does not affect plugin availability in apps like Audacity. Please feel free to close this @sadko4u.

**Comment by RoestVrijStaal** [comment]
Again, my previous question is not answered. "Just don't install" is not an answer.

Plugins ain't applications. Simple as that. .desktop files shouldn't be there from the very beginning.

**Comment by sadko4u** [comment]
@RoestVrijStaal 
I don't understand what's unclear for you. XDG integraion icons come with JACK standalone version of plugins. If you don't want to use standalone version of plugins, just don't install JACK version of bundle and you will get rid of flood.
Such DE's like GNOME don't follow the XDG standard and do not group plugins into a single folder as it should be. That's why you have such 'icon flooding' effect.

**Comment by matezoltanfarkas** [comment]
I would like to reopen this issue.

Since easyeffects has been ported to Qt (or maybe even before that), it soft-depends on these plugins.

This spam is really visible and disturbing on KDE. Extending the `NotShowIn=GNOME;` entry with KDE (as in `NotShowIn=GNOME;KDE`) fixes the issue.

A single 

```for file in $(ls | grep in.lsp_plug.lsp_plugins); do sed -i "s/NotShowIn=GNOME;/NotShowIn=GNOME;KDE/g" $file; done```

does the job locally.

**Comment by matezoltanfarkas** [comment]
could we get a fix in the same spirit as in https://github.com/lsp-plugins/lsp-plugins/issues/285 @sadko4u ?

**Comment by sadko4u** [comment]
Hello! Doesn't your KDE version group plugins into a single sub-menu?

**Comment by matezoltanfarkas** [comment]
No it doesn't.
It just vomits everything into the category "Multimedia".

**Comment by sadko4u** [comment]
What's your version of KDE and Linux distribution?

**Comment by matezoltanfarkas** [comment]
Arch Linux up to date KDE Plasma 6.
I can only tell you more once I'm home... ^^

**Comment by matezoltanfarkas** [comment]
Here's my `kinfo` output @sadko4u :

> Operating System: Arch Linux 
KDE Plasma Version: 6.5.3
KDE Frameworks Version: 6.20.0
Qt Version: 6.10.0
Kernel Version: 6.17.8-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 20 × 13th Gen Intel® Core™ i7-13700H
Memory: 64 Gio of RAM (62.5 Gio usable)
Graphics Processor 1: Intel® Iris® Xe Graphics
Graphics Processor 2: NVIDIA GeForce RTX 4060 Laptop GPU

Another unmentioned aspect of the issue that even packing the .desktop files into a subfolder causes them to pop-up in the searches quite agressively (e.g. the user gets almost 100% results of lsp-plugins if he/she searches for "A"), which is quite distracting and even disturibing.

**Comment by matezoltanfarkas** [comment]
the new release suffers from the same issue.

any ideas/progress @sadko4u ?

**Comment by sadko4u** [comment]
I didn't do anything related to this thread yet. The best solution probably could be to get rid of desktop entries and use a single launcher: https://github.com/lsp-plugins/lsp-plugins/issues/195

**Comment by matezoltanfarkas** [comment]
would it still be possible to extend the `NotShowIn=GNOME;` entry in the desktop files to `NotShowIn=GNOME;KDE;` in the meantime as a quick fix for users though?

That'd already be a huge help for audio fans!

**Comment by sadko4u** [comment]
How do you get the Menu being flattened?
I have a KDE 6 installation on my newly installed openSUSE Leap 16.0 and all seems to be right with menu:

<img width="874" height="1080" alt="Image" src="https://github.com/user-attachments/assets/27d3bb61-e263-4b39-af59-abbd4412decc" />

**Comment by matezoltanfarkas** [comment]
Interesting. In the std kickoff view, the "LSP Plugins" subfolder does not exist under the Multimedia entry:

<img width="748" height="608" alt="Image" src="https://github.com/user-attachments/assets/cf873ccb-f568-4ffb-afc4-e8693d85b5c0" />

**Comment by matezoltanfarkas** [comment]
and it still pops up in the krunner searches in kde, whenever and whatever the user is searching for something ^^

**Comment by oyvindln** [comment]
I got the same thing happening on kubuntu 25.10 (with up to date kde from kubuntu backports ppa). The "use compact list item style" setting doesn't seem to make any difference.

Same thing on my laptop running Opensuse TW. Maybe leap 16 used an older version of KDE where things worked differently.

Granted easyeffect doesn't seem to have a hard requirement on the specific plugin package that contains the desktop file on either distro, at least for doing basic eq so maybe it's less of an issue than on arch.

**Comment by laundmo** [comment]
> and it still pops up in the krunner searches in kde, whenever and whatever the user is searching for something

This, to me, is the biggest issue. Luckily on arch i could uninstall `lsp-plugins lsp-plugins-standalone` without uninstalling the various extension formats which don't have desktop files. It seems that the default `lsp-plugins` package depends on `lsp-plugins-standalone`

Alas, this is a workaround.

**Comment by sadko4u** [comment]
The upcoming release will solve the problem.
